packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,197 @@
+/// [previousFocus], and [focusInDirection] functions will perform these
+/// navigation operations on the focus tree.
+///
+/// The policy for what the "next" or "previous" widget is, or the widget in a
+/// particular direction, is determined by the [FocusTraversalPolicy] in force.
+///
+/// The policy that's in force is determined by looking up the widget hierarchy
+/// for a [DefaultFocusTraversal] widget, and obtaining the focus traversal
+/// policy from it. Different focus nodes can have difference policies, so part
+/// of your app can go in widget order, and part can go in reading order,
+/// depending upon the use case.
+///
+/// Predefined policies include [WidgetOrderFocusTraversalPolicy],
+/// [ReadingOrderTraversalPolicy], and [DirectionalFocusTraversalPolicyMixin],
+/// but you can also build your own policies based upon these policies.
 ///
 /// See also:
 ///
-///  * [FocusScopeNode], which is an interior node in the focus tree.
-///  * [FocusScope.of], which provides the [FocusScopeNode] for a given
-///    [BuildContext].
-class FocusNode extends ChangeNotifier {
-  FocusScopeNode _parent;
+///   * [Focus], a widget that manages a [FocusNode] and provides access to
+///     focus information and actions to its descendant widgets.
+///   * [FocusScope], a widget that manages a [FocusScopeNode] and provides
+///     access to scope information and actions to its descendant widgets.
+///   * [FocusManager], a singleton that manages the focus and distributes key
+///     events to focused nodes.
+///   * [FocusTraversalPolicy], an object used to determine how to move the
+///     focus to other nodes.
+///   * [DefaultFocusTraversal], a widget used to configure the default focus
+///     traversal policy for a widget subtree.
+class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
+  /// Creates a Focus node.
+  ///
+  /// All parameters are optional.
+  FocusNode({
+    String debugLabel,
+    this.onKey,
+  }) {
+    // Set it this way so that it does nothing on release builds.
+    this.debugLabel = debugLabel;
+  }
+
+  /// The context that was supplied to [reparentIfNeeded].
+  ///
+  /// This is typically the context in the build method for the widget that is
+  /// being focused, as it is used to determine the bounds of the widget.
+  BuildContext get context => _context;
+  BuildContext _context;
+
+  /// Called if this focus node receives a key event while focused (i.e. when
+  /// [hasFocus] returns true).
+  ///
+  /// It doesn't need to be the primary focus to receive events: it will also
+  /// receive key events if it is the ancestor of the node with the primary
+  /// focus (i.e. the one that returns true for [hasPrimaryFocus]), unless the
+  /// primary focus or one of its closer ancestors has returned true from its
+  /// [onKey], indicating that it handled the event.
+  ///
+  /// {@macro flutter.widgets.focus_manager.focus.keyEvents}
```

Comment:
this macro seems redundant with the earlier text. might be worth not including it?

Reply:
Hmm.  Yeah, I guess they are. It think I'll just nuke the paragraph above, and leave the macro paragraph.